presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/SetOperationNodeTranslator.java

```diff
@@ -39,31 +42,37 @@
 import java.util.Optional;
 
 import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
 import static com.google.common.collect.Iterables.concat;
 import static io.prestosql.spi.type.BigintType.BIGINT;
 import static io.prestosql.spi.type.BooleanType.BOOLEAN;
 import static io.prestosql.sql.analyzer.TypeSignatureProvider.fromTypes;
 import static io.prestosql.sql.analyzer.TypeSignatureTranslator.toSqlType;
 import static io.prestosql.sql.planner.plan.AggregationNode.singleGroupingSet;
 import static io.prestosql.sql.tree.BooleanLiteral.TRUE_LITERAL;
+import static io.prestosql.sql.tree.FrameBound.Type.CURRENT_ROW;
+import static io.prestosql.sql.tree.FrameBound.Type.UNBOUNDED_PRECEDING;
+import static io.prestosql.sql.tree.WindowFrame.Type.RANGE;
 import static java.util.Objects.requireNonNull;
 
 public class SetOperationNodeTranslator
 {
     private static final String MARKER = "marker";
     private final SymbolAllocator symbolAllocator;
     private final PlanNodeIdAllocator idAllocator;
-    private final ResolvedFunction countAggregation;
+    private final ResolvedFunction countFunction;
+    private final ResolvedFunction rowNumberFunction;
 
     public SetOperationNodeTranslator(Metadata metadata, SymbolAllocator symbolAllocator, PlanNodeIdAllocator idAllocator)
     {
         this.symbolAllocator = requireNonNull(symbolAllocator, "SymbolAllocator is null");
         this.idAllocator = requireNonNull(idAllocator, "PlanNodeIdAllocator is null");
         requireNonNull(metadata, "metadata is null");
-        this.countAggregation = metadata.resolveFunction(QualifiedName.of("count"), fromTypes(BOOLEAN));
+        this.countFunction = metadata.resolveFunction(QualifiedName.of("count"), fromTypes(BOOLEAN));
+        this.rowNumberFunction = metadata.resolveFunction(QualifiedName.of("row_number"), ImmutableList.of());
     }
 
-    public TranslationResult makeSetContainmentPlan(SetOperationNode node)
+    public TranslationResult makeSetContainmentPlanWithAggregation(SetOperationNode node)
```

Comment:
The names withAggregation and withoutAggregation are not very clear about the intent. Maybe rename to `makeSetContainmentPlanForDistinct` and `makeSetContainmentPlanForAll`